zstd-decompress.c

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2024, Stefan Hansson
+ */
+
+/* For memfd_create */
+#define _GNU_SOURCE
+
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <zstd.h>
+
+#include "zstd-decompress.h"
+
+/**
+ * zstd_decompress_file() - decompress a zstd-compressed file
+ * @filename:	path to a file to decompress
+ *
+ * Return: opened fd on success, -1 on error
+ */
+int zstd_decompress_file(const char *filename)
+{
+	/* Figure out the size of the file. */
+	struct stat file_stat;
+	if (stat(filename, &file_stat) == -1) {
+		perror("stat failed");
+		return -1;
+	}
+
+	const off_t file_size = file_stat.st_size;
+	const size_t size_t_file_size = (size_t)file_size;
```

Comment:
Why do you need to encode type into var name?

Reply:
Really only because I couldn't reuse `file_size` and I couldn't come up with a better name. Do you have any suggestions?

Reply:
can you set `size_t file_size = file_stat.st_size` ?